File: tests/misc/order.py
def left():
    print 'left'
    return 0

def right():
    print 'right'
    return 0

print "left() + right()"
left() + right()
print

def call(a, b):
    pass
print "call(left(), right())"
call(left(), right())
print

def index():
    print 'index'
    return 0

def value():
    print 'value'
    return 0

def array():
    print 'array'
    return [0]

print "array()[index()]"
array()[index()]
print

print "array()[index()] = value()"
array()[index()] = value()
print

File: tests/misc/order.c
#include <stdio.h>

int yindex()
{
    puts("index");
    return 0;
}

int value()
{
    puts("value");
    return 0;
}

int *array()
{
    static int a[] = {0};
    puts("array");
    return a;
}

int left()
{
    puts("left");
    return 0;
}

int right()
{
    puts("right");
    return 0;
}

void add(int a, int b)
{
}

int main()
{
    array()[yindex()] = value();
    add(left(), right());
    return 0;
}
